doc/integrator/extend_application.rst

```diff
@@ -40,17 +40,243 @@ In the ``vars.yaml`` file your interface should be declared like that:
+      - --paste
+      - development.ini
+    volumes:
+      - ./custom/custom:/app/custom
+
+.. note::
+
+   If you needs the user credentials, you can do:
+
+   .. code:: python
+
+      requests.get(
+          "http://geoportal:8080/loginuser",
+          headers={"Cookie": request.headers.get("Cookie"), "Referer": request.referrer},
+      ).json()
 
 Custom JavaScript file
 ----------------------
 
-Will be filled later.
+In this example we will add a button in the tools bar, that open tool panel, that can be used to send a feedback.
+
+The tool button should be an instance of
+`gmfapi.elements.ToolButtonElement<https://camptocamp.github.io/ngeo/|main_branch|/apidoc/classes/srcapi_elements_ToolButtonElement.default.html>`_.
+
+In this example he will directly use tool button will be an instance of
+`gmf-tool-button<https://camptocamp.github.io/ngeo/|main_branch|/apidoc/classes/srcapi_elements_ToolButtonElement.ToolButtonDefault.html>`_.
+
+Then we will includes the following HTML in the canvas element:
+
+```html
+<gmf-tool-button slot="tool-button" iconClasses="fas fa-file-signature" panelName="feedback"></gmf-tool-button>
+```
+
+The panel will be included with the following HTML:
+
+```html
+<proj-feedback slot="tool-panel-feedback"></proj-feedback>
+```
+
+And should should be an instance of
+`gmfapi.elements.ToolPanelElement<https://camptocamp.github.io/ngeo/|main_branch|/apidoc/classes/srcapi_elements_ToolPanelElement.default.html>`_.
+
+
+In this tutorial we will create a new WebComponent based on `Lit <https://lit.dev/>`_,
+and build by `Vite <https://vitejs.dev/>`_. We will directly get the component and the build environment
+from the demo:
+
+.. prompt:: bash
+
+   cd /tmp
+   git clone git@github.com:camptocamp/demo_geomapfish.git
+   cd -
+   mv /tmp/demo_geomapfish/webcomponents \
+      /tmp/demo_geomapfish/package.json \
+      /tmp/demo_geomapfish/package-lock.json \
+      /tmp/demo_geomapfish/tsconfig.json \
+      /tmp/demo_geomapfish/vite.config.ts .
+
+Add tho following lines in the ``.dockerignore``:
+
+.. code::
+
+   !webcomponents/
+   !package.json
+   !package-lock.json
+   !tsconfig.json
+   !vite.config.ts
+
+Add tho following lines in the ``.gitignore``:
+
+.. code::
+
+   /node_modules
+
+Add tho following lines at the end of ``Dockerfile``:
+
+.. code::
+
+   ###############################################################################
+
+   FROM node:16-slim AS custom-build
+
+   WORKDIR /app
+   COPY package.json ./
+
+   RUN npm install
+
+   COPY tsconfig.json vite.config.ts ./
+   COPY webcomponents/ ./webcomponents/
+   RUN npm run build
+
+   ###############################################################################
+
+   FROM gmf_config AS config
+   COPY --from=custom-build /app/dist/ /etc/geomapfish/static/custom/
+
+Apply the following diff in the ``geoportal/vars.yaml``:
+
+.. code:: diff
+
+   +
+   +         # For dev, the corresponding values in static should also be removed.
```

Comment:
I suppose this goes in interfaces_config/desktop for example.
This might not be trivial for everybody.

```suggestion
   +         # For dev, the corresponding values in static should also be uncommented.
```

Reply:
=> commented :-)